app/src/main/java/de/westnordost/streetcomplete/data/osm/mapdata/MapDataCache.kt

```diff
@@ -383,8 +386,13 @@ class MapDataCache(
                 return result
             }
 
-            // get nodes again, this contains the newly added nodes, but maybe not the old ones if cache was trimmed
-            nodes.addAll(spatialCache.get(bbox))
+            // add newly fetched nodes from elements
+            // getting nodes from spatialCache can cause issues, as tiles in the bbox may now be removed unexpectedly
```

Comment:
how/when though? This is all synchronized, isn't it?

Reply:
Updating the cache may remove tiles, see comment below for an explanation.